src/newcharacter.cpp

```diff
@@ -1137,10 +1172,11 @@ int set_profession(WINDOW *w, player *u, int &points)
         fold_and_print(w_description, 0, 0, FULL_SCREEN_WIDTH - 2, c_green,
                        sorted_profs[cur_id]->description(u->male));
 
-        calcStartPos(iStartPos, cur_id, iContentHeight, sorted_profs.size());
+        calcStartPos(iStartPos, cur_id, iContentHeight, profs_length);
         //Draw options
-        for (int i = iStartPos; i < (int)iStartPos + ((iContentHeight > (int)sorted_profs.size()) ?
-                (int)sorted_profs.size() : (int)iContentHeight); i++) {
+        int i;
+        for (i = iStartPos; i < (int)iStartPos + ((iContentHeight > (int)profs_length) ?
+                (int)profs_length : (int)iContentHeight); i++) {
```

Comment:
That huge `if` is unreadable. Could the middle of it be yanked out?
I mean something like:
```
const int end_pos = (int)iStartPos + ((iContentHeight > (int)profs_length) ?
    (int)profs_length : (int)iContentHeight);
for( i = start_pos; i < end_pos; i++ )
```
Assuming that this `end_pos` doesn't change during the loop execution.